apiserver/facades/client/client/status.go

```diff
@@ -888,7 +899,7 @@ func paramsJobsFromJobs(jobs []state.MachineJob) []multiwatcher.MachineJob {
 
 func (context *statusContext) processApplications() map[string]params.ApplicationStatus {
 	applicationsMap := make(map[string]params.ApplicationStatus)
-	for _, s := range context.applications {
+	for _, s := range context.allAppsUnitsCharmBindings.applications {
```

Comment:
At the top of these functions where context.allAppsUnitsCharmBindings is used, to save repetition, you could just use a local variable:
appStatusInfo := context.allAppStatusInfo
...
if latestCharm, ok := appStatusInfo.latestCharms[...]; ok && latestCharm != nil {...}
etc